impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultDependencyResolver.java

```diff
@@ -68,6 +68,22 @@
 @Singleton
 public class DefaultDependencyResolver implements DependencyResolver {
 
+    /**
+     * Cache of information about the modules contained in a path element.
+     * This cache is created when first needed. It may be never created.
```

Comment:
```suggestion
```
imho no, return case will always call.

<img width="1233" alt="image" src="https://github.com/user-attachments/assets/bfd2b62e-0a7a-4f86-b40a-25c1ca5489bf" />

Reply:
> imho no, as always have return call, on default.

Right. Checking again, a see no code path where the cache would not be created. I will remove the lazy instantiation aspect.